geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/QueryPlanFilterVisitor.scala

```diff
@@ -143,12 +145,50 @@ protected class QueryPlanFilterVisitor(sft: SimpleFeatureType) extends Duplicati
     }
   }
 
+  private def handleInArrayEquals(filter: PropertyIsEqualTo, extraData: AnyRef): Option[Filter] = {
+    var inArrayFunction: InArrayFunction = null
+    var hasTrue: Boolean = false
+
+    def inspectExpression(expression: Expression) = {
+      expression match {
+        case function: InArrayFunction =>
+          inArrayFunction = function
+        case literal: Literal if literal.getValue.equals(java.lang.Boolean.TRUE) =>
+          hasTrue = true
+        case _ =>
+      }
+    }
+
+    inspectExpression(filter.getExpression1)
+    inspectExpression(filter.getExpression2)
+
+    if (inArrayFunction != null && hasTrue) {
+      val attribute = inArrayFunction.getParameters.get(0)
+      val array = inArrayFunction.getParameters.get(1)
+      import scala.collection.JavaConversions._
+      val filters = array.evaluate(null).asInstanceOf[util.List[_]].toSet[Any].map {
+        o => ff.equals(attribute, ff.literal(o))
```

Comment:
instead of using `ff` here and below, it would be good to use `val ff = getFactory(extraData)` right above, to allow users to pass in their own factory hints. I'm not sure that's even possible, given our workflow, but seems safer to follow the pattern used in the rest of the class

Reply:
Will do!